src/app/public/src/modules/shared/search-results-provider.ts

```diff
@@ -0,0 +1,143 @@
+import { Injectable } from '@angular/core';
+import { StacheHttp } from './stache-http.service';
+import { StacheConfigService } from './config.service';
+import { StacheSearchResult } from './search-result';
+import 'rxjs/add/operator/toPromise';
+
+@Injectable()
+export class StacheSearchResultsProvider {
+
+  private moreResultsUrl: string;
+  private moreResultsLabel: string = 'Show More';
+  private htmlFields: any = {
+    description: true
+  };
+  private isInternal: boolean = true;
+  private siteNames: string[] = [];
+  private endpoints: any = {
+    internal: 'query',
+    public: 'query-public'
+  };
+  private queryUrlBase: string = 'https://stache-search-query.sky.blackbaud.com';
+
+  constructor(
+    private http: StacheHttp,
+    private config: StacheConfigService
```

Comment:
I feel like we can eliminate a ton of the verbose lines of code.   I see only one place where we are actually using `this.config._____` without going all the way down into `config.skyux.appSettings.stache.searchConfig`.  Couldn't we just create a method like

```
validateSearchConfig(): any {
  let searchConfig = {};

  if (this.configService.skyux.appsettings 
    && this.configService.skyux.appSettings.stache
    && this.configService.skyux.appSettings.stache.searchConfig)  {
    searchConfig = this.configService.skyux.appSettings.stache.searchConfig;
  }

  this.searchConfig = searchConfig;
}
```

Reply:
Valid point.